plugin/adsk/plugin/exportTranslator.cpp

```diff
@@ -0,0 +1,211 @@
+        MDagPath dagPath;
+        if (objSelList.getDagPath(i, dagPath) == MS::kSuccess) {
+            dagPaths.insert(dagPath);
+        }
+    }
+    
+    if (dagPaths.empty()) {
+        TF_WARN("No DAG nodes to export. Skipping.");
+        return MS::kSuccess;
+    }
+
+    const std::vector<double> timeSamples = UsdMayaWriteUtil::GetTimeSamples(
+            timeInterval, std::set<double>(), frameStride);
+    PXR_NS::UsdMayaJobExportArgs jobArgs = PXR_NS::UsdMayaJobExportArgs::CreateFromDictionary(
+            userArgs, dagPaths, timeSamples);
+    for (unsigned int i=0; i < filteredTypes.length(); ++i) {
+        jobArgs.AddFilteredTypeName(filteredTypes[i].asChar());
+    }
+
+    UsdMaya_WriteJob writeJob(jobArgs);
+    if (!writeJob.Write(fileName, append)) {
+        return MS::kFailure;
+    }
+    
+    return MS::kSuccess;
+}
+
+MPxFileTranslator::MFileKind
+UsdMayaExportTranslator::identifyFile(
+        const MFileObject& file,
+        const char*  /*buffer*/,
+        short  /*size*/) const
+{
+    MFileKind retValue = kNotMyFileType;
+    const MString fileName = file.fullName();
+    const int lastIndex = fileName.length() - 1;
+
+    const int periodIndex = fileName.rindex('.');
+    if (periodIndex < 0 || periodIndex >= lastIndex) {
+        return retValue;
+    }
+
+    const MString fileExtension = fileName.substring(periodIndex + 1, lastIndex);
+
+    if (fileExtension ==
+            UsdMayaTranslatorTokens->UsdFileExtensionDefault.GetText() || 
+        fileExtension ==
+            UsdMayaTranslatorTokens->UsdFileExtensionASCII.GetText() || 
+        fileExtension ==
+            UsdMayaTranslatorTokens->UsdFileExtensionCrate.GetText() ||
+        fileExtension ==
+            UsdMayaTranslatorTokens->UsdFileExtensionPackage.GetText()) {
+        retValue = kIsMyFileType;
+    }
+
+    return retValue;
+}
+
+/* static */
+const std::string&
+UsdMayaExportTranslator::GetDefaultOptions()
+{
+    static std::string defaultOptions;
+    static std::once_flag once;
+    std::call_once(once, []() {
+        std::vector<std::string> entries;
+        for (const std::pair<std::string, VtValue> keyValue :
+                PXR_NS::UsdMayaJobExportArgs::GetDefaultDictionary()) {
+            if (keyValue.second.IsHolding<bool>()) {
+                entries.push_back(TfStringPrintf("%s=%d",
+                        keyValue.first.c_str(),
+                        static_cast<int>(keyValue.second.Get<bool>())));
+            }
+            else if (keyValue.second.IsHolding<std::string>()) {
+                entries.push_back(TfStringPrintf("%s=%s",
+                        keyValue.first.c_str(),
+                        keyValue.second.Get<std::string>().c_str()));
+            }
+        }
+        entries.push_back("animation=0");
+        entries.push_back("startTime=1");
+        entries.push_back("endTime=1");
+        entries.push_back("frameStride=1.0");
+        defaultOptions = TfStringJoin(entries, ";");
```

Comment:
Lordy loo! Talk about convoluted! Just use a ```std::ostringstream```, and build the string in one go. Better yet, use our code generators that handle all of this boiler plate stuff for you ;)

Reply:
Yeah, I guess instead of a vector of strings with a join at the end we could use ostringstream.  I’ll take a look at simplifying this one.

One general comment to make about using any of the AL utility code…  At this point we have moved a lot of the Pixar translator code down into the core lib, so pretty much all of the options defined down there are still exactly how Pixar has had them, and this Default Dictionary stuff is all Pixar right now, but not much of the AL code has been moved there yet.  I don’t think anyone is avoiding using AL code but a lot it isn’t available just yet in the shared area.